rbc/irtools.py

```diff
@@ -338,31 +380,87 @@ def compile_to_LLVM(functions_and_signatures,
         flags.set('no_cfunc_wrapper')
 
     function_names = []
+    succesful_fids = []
+    required_libraries = set()
     for func, signatures in functions_and_signatures:
-        for sig in signatures:
+        for fid, sig in signatures.items():
             fname = func.__name__ + sig.mangling()
-            function_names.append(fname)
             args, return_type = sigutils.normalize_signature(
                 sig.tonumba(bool_is_int8=True))
-            cres = compiler.compile_extra(typingctx=typing_context,
-                                          targetctx=target_context,
-                                          func=func,
-                                          args=args,
-                                          return_type=return_type,
-                                          flags=flags,
-                                          library=main_library,
-                                          locals={},
-                                          pipeline_class=pipeline_class)
+            try:
+                cres = compiler.compile_extra(typingctx=typing_context,
+                                              targetctx=target_context,
+                                              func=func,
+                                              args=args,
+                                              return_type=return_type,
+                                              flags=flags,
+                                              library=main_library,
+                                              locals={},
+                                              pipeline_class=pipeline_class)
+            except OmnisciUnsupportedError as msg:
+                for m in re.finditer(r'[|]OmnisciUnsupportedError[|](.*?)\n', str(msg), re.S):
+                    print(f'Skipping {fname} `{sig}`: {m.group(0)[25:]}')
+                continue
+            except nb_errors.TypingError as msg:
+                for m in re.finditer(r'[|]OmnisciUnsupportedError[|](.*?)\n', str(msg), re.S):
+                    print(f'Skipping {fname} `{sig}`: {m.group(0)[25:]}')
+                    break
+                else:
+                    raise
+                continue
+            except Exception:
+                raise
+
+            skip = False
+            intrinsics, declarations, defined, libraries = get_called_functions(
+                cres.library, cres.fndesc.llvm_func_name)
+            for f in declarations:
+                if target.is_gpu:
+                    if f.startswith('__nv_') and f[5:] in get_libdevice_functions():
+                        # numba supports libdevice functions starting from 0.52
+                        continue
```

Comment:
You should also check if the server version is >= (5, 5). Although I do not think we have access to the server version at this program point.

Reply:
`retrieve_targets` now covers this with
```
            elif target_info.is_gpu and self.version >= (5, 5):
                target_info.add_library('libdevice')
```

Also, `target_info.software` now returns `('OmnisciDB', (5, 5, 0, 'dev...'))`